File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, arr: List[int]) -> int:
        n=len(arr)
        maxi=arr[0]
        res=arr[0]
        for i in range(1,n):
            maxi=max(maxi+arr[i],arr[i])
            res=max(res,maxi)
        return res

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] arr) {
        int n=arr.length;
        int res=arr[0];
        int maxi=arr[0];
        for (int i=1;i<n;i++){
            maxi=Math.max(maxi+arr[i],arr[i]);
            res=Math.max(maxi,res);
        }
        return res;
    }
}